falcon/constants.py

```diff
@@ -1,5 +1,13 @@
 from enum import Enum
 import os
+import sys
+
+
+PYPY = (sys.implementation.name == 'pypy')
```

Comment:
I went ahead and factored out this constant since we were using it in multiple places.